src/parameterizations/lateral/MOM_Zanna_Bolton.F90

```diff
@@ -613,6 +645,145 @@ subroutine compute_stress(G, GV, CS)
+  type(verticalGrid_type), intent(in)    :: GV   !< The ocean's vertical grid structure
+  type(ZB2020_CS),         intent(inout) :: CS   !< ZB2020 control structure.
+
+  integer :: is, ie, js, je, Isq, Ieq, Jsq, Jeq, nz
+  integer :: i, j, k, n, m
+  integer :: ii, jj
+  integer :: nij
+
+  real, allocatable :: x(:,:)        ! Vector of non-dimensional input features
+                                     ! number of horizontal grid points x
+                                     ! (sh_xy, sh_xx, vort_xy) on a stencil    [nondim]
+  real, allocatable :: y(:,:)        ! Vector of nondimensional
+                                     ! output features number of horizontal grid points x
+                                     ! (Txy,Txx,Tyy) [nondim]
+  real :: yy(3)                      ! Vector of dimensional
+                                     ! output features (Txy,Txx,Tyy) [L2 T-2 ~> m2 s-2]
+  real :: input_norm                 ! Norm of input features [T-1 ~> s-1]
+  real :: tmp                        ! Temporal value of squared norm [T-2 ~> s-2]
+  integer :: offset                  ! Half the stencil size. Used for selection
+  integer :: stencil_points          ! The number of points after flattening
+
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)) :: &
+        sh_xy_h,   & ! sh_xy interpolated to the center [T-1 ~> s-1]
+        vort_xy_h, & ! vort_xy interpolated to the center [T-1 ~> s-1]
+        norm_h       ! Norm of input feautres in center points [T-1 ~> s-1]
+
+  real, dimension(SZI_(G),SZJ_(G)) :: &
+        sqr_h, & ! Squared norm of velocity gradients in center points [T-2 ~> s-2]
+        Txy      ! Predicted Txy in center points                      [T-1 ~> s-1]
+
+  call cpu_clock_begin(CS%id_clock_stress_ANN)
+
+  is  = G%isc  ; ie  = G%iec  ; js  = G%jsc  ; je  = G%jec ; nz = GV%ke
+  Isq = G%IscB ; Ieq = G%IecB ; Jsq = G%JscB ; Jeq = G%JecB
+
+  ! Number of horizontal grid points in ANN inference loop below
+  nij = (ie - is + 5) * (je - js + 5)
+  allocate(x(nij, 3 * CS%stencil_size**2))
+  allocate(y(nij, 3))
+
+  sh_xy_h = 0.
+  vort_xy_h = 0.
+  norm_h = 0.
+
+  call pass_var(CS%sh_xy, G%Domain, clock=CS%id_clock_mpi, position=CORNER)
+  call pass_var(CS%sh_xx, G%Domain, clock=CS%id_clock_mpi)
+  call pass_var(CS%vort_xy, G%Domain, clock=CS%id_clock_mpi, position=CORNER)
+
+  offset = (CS%stencil_size-1)/2
+  stencil_points = CS%stencil_size**2
+
+  ! Interpolate input features
+  do k=1,nz
+    do j=js-2,je+2 ; do i=is-2,ie+2
+      ! It is assumed that B.C. is applied to sh_xy and vort_xy
+      sh_xy_h(i,j,k) = 0.25 * ( (CS%sh_xy(I-1,J-1,k) + CS%sh_xy(I,J,k)) &
+                              + (CS%sh_xy(I-1,J,k) + CS%sh_xy(I,J-1,k)) )
+
+      vort_xy_h(i,j,k) = 0.25 * ( (CS%vort_xy(I-1,J-1,k) + CS%vort_xy(I,J,k)) &
+                                + (CS%vort_xy(I-1,J,k) + CS%vort_xy(I,J-1,k)) )
+
+      sqr_h(i,j) = (CS%sh_xx(i,j,k)**2 + sh_xy_h(i,j,k)**2 + vort_xy_h(i,j,k)**2) * G%mask2dT(i,j)
```

Comment:
Please add parentheses to specify the order of arithmetic with the sums on line 724 so that we are more likely to get the same answers with different compilers or compiler settings.